validator/src/main.rs

```diff
@@ -392,6 +392,20 @@ fn hash_validator(hash: String) -> Result<(), String> {
         .map_err(|e| format!("{:?}", e))
 }
 
+/// Returns the default shred storage size (include both data and coding
+/// shreds) based on the validator config.
+fn default_fifo_shred_storage_size(vc: &ValidatorConfig) -> u64 {
+    // This is an estimation that is slightly larger than average so that we
+    // can make sure FIFO rarely evits data/coding shreds that is younger
+    // than the configured max_ledger_shreds while avoiding allocating too
+    // much disk space for data/coding shreds.
+    const AVG_SHRED_SIZE_BYTE: u64 = 3200;
+    match vc.max_ledger_shreds {
+        Some(max_ledger_shreds) => (max_ledger_shreds / 2) * AVG_SHRED_SIZE_BYTE,
+        None => DEFAULT_ROCKS_FIFO_SHRED_STORAGE_SIZE_BYTES,
```

Comment:
Given how validator is written, `vc.max_ledger_shreds` should always be a `Some(...)`; wondering if we should rework this.

Also wondering if `DEFAULT_ROCKS_FIFO_SHRED_STORAGE_SIZE_BYTES` should match the size that would result from the estimate here with `DEFAULT_MAX_LEDGER_SHREDS`

Reply:
On top of this PR, the DEFAULT_ROCKS_FIFO_SHRED_STORAGE_SIZE_BYTES will be removed in https://github.com/solana-labs/solana/pull/27965.  I've thinking about merging these two PRs, but I feel it's probably better to have one PR for one purpose: this PR focuses on making FIFO arguments public, and the other PR focuses on deprecate unused default.

Reply:
Ok, fair enough about the other PR addressing the default.